sdk/src/account/record.rs

```diff
@@ -51,12 +60,80 @@ impl RecordCiphertext {
     }
 }
 
+impl Deref for RecordCiphertext {
+    type Target = RecordCiphertextNative;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl From<RecordCiphertextNative> for RecordCiphertext {
+    fn from(record_ciphertext: RecordCiphertextNative) -> Self {
+        Self(record_ciphertext)
+    }
+}
+
 #[pyclass(frozen)]
 pub struct RecordPlaintext(RecordPlaintextNative);
 
 #[pymethods]
 impl RecordPlaintext {
+    /// Reads in the plaintext string.
+    #[staticmethod]
+    fn from_string(s: &str) -> anyhow::Result<Self> {
+        Ok(Self::from(RecordPlaintextNative::from_str(s)?))
+    }
+
+    /// Returns the owner of the record as a string
+    fn owner(&self) -> String {
+        self.0.owner().to_string()
+    }
+
+    /// Returns the nonce of the record as a string
+    fn nonce(&self) -> String {
+        self.0.nonce().to_string()
+    }
+
+    /// Attempt to get the serial number of a record to determine whether or not is has been spent
+    pub fn serial_number_string(
+        &self,
+        private_key: &PrivateKey,
+        program_id: &str,
```

Comment:
I would like sdk to be typed as much as possible and to have proper ProgramID and Identifier types. This is not hard to do, I can do it in a separate PR. Would you create a list of functions/types you want to add to sdk?

Reply:
@kpp You're %100 right. I'd love to have contributions to the SDK that make the ProgramID and Identifier types. We will eventually get to these, but if you get to them first, we will not stop you!

A short list of the types we'd want are:
-- Program Types --
* Identifier
* Process
* Program
* ProgramID
* ProvingKey
* VerifyingKey

-- Data Types --
* RecordPlaintext
* RecordCipherText

-- Ledger Types
* Transaction
* Transition

Reply:
Almost all of it is done in https://github.com/AleoHQ/python-sdk/pull/22